python/pyspark/mllib/clustering.py

```diff
@@ -86,6 +88,84 @@ def train(cls, rdd, k, maxIterations=100, runs=1, initializationMode="k-means||"
         return KMeansModel([c.toArray() for c in centers])
 
 
+class GaussianMixtureModel(object):
+
+    """A clustering model derived from the Gaussian Mixture Model method.
+
+    >>> clusterdata_1 =  sc.parallelize(array([-0.1,-0.05,-0.01,-0.1,
+    ...                                         0.9,0.8,0.75,0.935,
+    ...                                        -0.83,-0.68,-0.91,-0.76 ]).reshape(6,2))
+    >>> model = GaussianMixtureEM.train(clusterdata_1, 3, convergenceTol=0.0001,
+    ...                                 maxIterations=50, seed=10)
+    >>> labels = model.predict(clusterdata_1).collect()
+    >>> labels[0]==labels[1]
+    False
+    >>> labels[1]==labels[2]
+    True
+    >>> labels[4]==labels[5]
+    True
+    >>> clusterdata_2 =  sc.parallelize(array([-5.1971, -2.5359, -3.8220,
+    ...                                        -5.2211, -5.0602,  4.7118,
+    ...                                         6.8989, 3.4592,  4.6322,
+    ...                                         5.7048,  4.6567, 5.5026,
+    ...                                         4.5605,  5.2043,  6.2734]).reshape(5,3))
+    >>> model = GaussianMixtureEM.train(clusterdata_2, 2, convergenceTol=0.0001,
+    ...                                 maxIterations=150, seed=10)
+    >>> labels = model.predict(clusterdata_2).collect()
+    >>> labels[0]==labels[1]==labels[2]
+    True
+    >>> labels[3]==labels[4]
+    True
+    """
+
+    def __init__(self, weights, gaussians):
+        self.weights = weights
+        self.gaussians = gaussians
+        self.k = len(self.weights)
+
+    def predict(self, X):
+        """
+        Find the cluster to which the points in X has maximum membership
+        in this model.
+        Returns an RDD of cluster labels.
+        """
+        cluster_labels = self.predictSoft(X).map(lambda x: x.index(max(x)))
+        return cluster_labels
+
+    def predictSoft(self, X):
+        """
+        Find the membership of each point in X to all mixture components.
+        Returns an RDD of array of double values.
+        """
+        means_temp = ()
+        sigmas_temp = ()
+        for i in range(self.k):
+            means_temp = means_temp + (self.gaussians[i].mu,)
+            sigmas_temp = sigmas_temp + (self.gaussians[i].sigma,)
+        membership_matrix = callMLlibFunc("predictGMM", X.map(_convert_to_vector),
+                                          self.weights, means_temp, sigmas_temp)
+        return membership_matrix
+
+
+class MultiVariateGaussian(object):
```

Comment:
This class should go to `mllib.stat.distribution` and we can declare it as a namedtuple. Note that we use `Multivariate` but not `MultiVariate`.

This might require some refactoring of `stat.py`. I'm going to make a PR and ping you back.